Jack/Property.fs

```diff
@@ -2,83 +2,153 @@
 
 open FSharpx.Collections
 
-type Result =
-    | Failure of List<string>
-    | Success
+type Report =
+    | Report of List<string>
+
+type Result<'a> =
+    | Failure
     | Discard
+    | Success of 'a
+
+type Property<'a> =
+    | Property of Gen<Report * Result<'a>>
```

Comment:
It was just for failure messages, but it could be used to support #25 or any other types of things that we might want to track during a test run.